packages/web/src/common/store/pages/track/sagas.js

```diff
@@ -172,7 +173,7 @@ function* watchFetchTrack() {
           yield put(pushRoute(NOT_FOUND_PAGE))
           return
         }
-      } else {
+      } else if (!skipSideEffects) {
```

Comment:
Can you explain a bit what was happening here? I'm curious

Reply:
- when you unlock, say, a follow gated track, all follow gated tracks by that user will be unlocked for you.
- as we poll these tracks for their premium content signatures for the user, these "side effects" would be called.
- these side effects set the track id, which we don't want as they sometimes update the page with the wrong track's data, causing a very bad looking bug. e.g. tracks X and Y are follow gated, im on track X's page, but now i keep seeing the track title flicker between X and Y
- the side effects also add the track to lineup, which we also don't care about in this case

Reply:
It sounds like what you want to do is call `retrieveTracks`? Is there a reason we call `fetchTrack` in the trackPage sagas instead?

Reply:
hmm, maybe i do want retrieveTracks, double-checking. thanks

Reply:
adding this to my upcoming items. will def look at it. thx again.
going to merge this for now